bundles/org.openhab.binding.openthermgateway/src/main/java/org/openhab/binding/openthermgateway/handler/OpenThermGatewayHandler.java

```diff
@@ -108,183 +101,184 @@ public void handleCommand(ChannelUID channelUID, Command command) {
+        for (Thing thing : getThing().getThings()) {
+            BaseDeviceHandler handler = (BaseDeviceHandler) thing.getHandler();
 
-        // retry connection if disconnect is not explicitly requested
-        if (!explicitDisconnect && conf != null && conf.connectionRetryInterval > 0) {
-            logger.debug("Scheduling to reconnect in {} seconds.", conf.connectionRetryInterval);
-            reconnectTask = scheduler.schedule(this::connect, conf.connectionRetryInterval, TimeUnit.SECONDS);
+            if (handler != null) {
+                handler.receiveMessage(message);
+            }
         }
     }
 
     @Override
-    public void receiveMessage(Message message) {
-        if (DataItemGroup.dataItemGroups.containsKey(message.getID())) {
-            DataItem[] dataItems = DataItemGroup.dataItemGroups.get(message.getID());
-
-            for (DataItem dataItem : dataItems) {
-                String channelId = dataItem.getSubject();
-
-                if (!OpenThermGatewayBindingConstants.SUPPORTED_CHANNEL_IDS.contains(channelId)
-                        || (dataItem.getFilteredCode() != null && dataItem.getFilteredCode() != message.getCode())) {
-                    continue;
-                }
-
-                State state = null;
-
-                switch (dataItem.getDataType()) {
-                    case FLAGS:
-                        state = OnOffType.from(message.getBit(dataItem.getByteType(), dataItem.getBitPos()));
-                        break;
-                    case UINT8:
-                    case UINT16:
-                        state = new DecimalType(message.getUInt(dataItem.getByteType()));
-                        break;
-                    case INT8:
-                    case INT16:
-                        state = new DecimalType(message.getInt(dataItem.getByteType()));
-                        break;
-                    case FLOAT:
-                        float value = message.getFloat();
-                        @Nullable
-                        Unit<?> unit = dataItem.getUnit();
-                        state = (unit == null) ? new DecimalType(value) : new QuantityType<>(value, unit);
-                        break;
-                    case DOWTOD:
-                        break;
-                }
+    public void connectionStateChanged(ConnectionState state) {
+        scheduler.submit(() -> connectionStateChangedTask(state));
+    }
 
-                if (state != null) {
-                    logger.debug("Received update for channel '{}': {}", channelId, state);
-                    updateState(channelId, state);
-                }
+    private void connectionStateChangedTask(ConnectionState state) {
+        if (this.state != state) {
+            this.state = state;
+
+            switch (state) {
+                case INITIALIZING:
+                    updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.NONE, "Initializing");
```

Comment:
Please provide i18n text for this as well as next two strings below.

Reply:
> Can you regenerate i18n texts?

Sorry but what do you mean by this?

Reply:
Simply to move the texts to the i18n file so they could be translated. But from comment below seems like it will not be needed. :)